lm_eval/models/neuron_optimum.py

```diff
@@ -580,15 +489,41 @@ def _collate(x):
                 cont_toks_list.append(continuation_enc)
                 inplens.append(inplen)
 
-            # create encoder attn mask and batched conts, if seq2seq
-            call_kwargs = {}
+            # Add dummy inputs up to the model static batch size
+            if len(inps) < self.batch_size:
+                inps = inps + [
+                    torch.zeros_like(inps[0]),
+                ] * (self.batch_size - len(inps))
+
+            masks = [torch.ones_like(inp) for inp in inps]
             batched_inps = lm_eval.models.utils.pad_and_concat(
                 padding_len_inp, inps, padding_side="right"
             )  # [batch, padding_len_inp]
 
-            multi_logits = F.log_softmax(
-                self._model_call(batched_inps, **call_kwargs), dim=-1
-            )  # [batch, padding_length (inp or cont), vocab]
+            batched_masks = lm_eval.models.utils.pad_and_concat(
+                padding_len_inp, masks, padding_side="right"
+            )
+            if self.model.model.neuron_config.output_all_logits:
```

Comment:
I turn this option on only for models exported with batch_size 1 in the `main` branch of `optimum-neuron` (with previous releases it is always off).
That's because:
- it has not been tested exhaustively, especially with models exported with continuous batching enabled, which is the default for bigger llms like llamas,
- it is probably slower to return all logits for batch_size >> 1 (to be confirmed).
To summzrize:
- you will automatically enter the condition if batch size is 1,
- if you use a higher batch size, you will get a slowdown because of the iterative evaluation, so it might not be worth it.